bin/trigger_changelog.py

```diff
@@ -0,0 +1,181 @@
+    prs = {}
+    for item in items:
+        pr_number = str(item['number'])
+        pr_url = item['pull_request']['html_url']
+        pr_title = item['title']
+        pr_query = Constants.GITHUB_API_URL + \
+            "/repos/shadow-robot/"+repo+"/pulls/"+pr_number
+        pr_query_result = requests.get(pr_query, auth=(
+            Constants.GIT_USERNAME, Constants.GIT_TOKEN))
+        pr = json.loads(pr_query_result.text)
+        pr_branch = pr['head']['ref']
+        pr_body = pr['body']
+        pr_title_continued = ""
+        try:
+            if type(pr_body) == str or type(pr_body) == bytes:
+                pr_title_continued = re.search(
+                    Constants.REGEXP_TITLE_CONTINUED, pr_body).group(1)
+        except AttributeError:
+            continue
+        pr_title += pr_title_continued
+        pr_title = pr_title.replace("…", "")
+        pr_title = pr_title.replace("## Proposed changes", "")
+        user_story = None
+        try:
+            user_story = re.search(
+                Constants.REGEXP_BRANCH_STORY, pr_branch).group(0)
+        except AttributeError:
+            pass
+        jira_link = "n/a"
+        if user_story is not None:
+            jira_link = "https://shadowrobot.atlassian.net/browse/"+user_story
+        pr_merged = item['closed_at']
+        prs[pr_url] = [pr_title, pr_branch, jira_link, pr_merged]
+    return prs
+
+
+def recursive_get_repos_from_rosinstall(owner, repo, branch, repo_dict):
+    repository_rosinstall_url = Constants.URL_BEGINNING + \
+        owner+'/'+repo+'/'+branch+'/'+Constants.REPOSITORY_ROSINSTALL
+    repository_rosinstall_result = requests.get(
+        repository_rosinstall_url, auth=(Constants.GIT_USERNAME, Constants.GIT_TOKEN))
+    if repository_rosinstall_result.status_code == 200:
+        raw_repos = repository_rosinstall_result.text.split("\n")
+        for raw_repo in raw_repos:
+            if raw_repo.startswith("    uri: "):
+                repo_url = raw_repo.replace("    uri: ", "")
+                repo_owner = re.search(
+                    Constants.REGEXP_OWNER_REPO, repo_url).group(1)
+                if repo_owner == 'shadow-robot':
+                    repo_name = re.search(
+                        Constants.REGEXP_OWNER_REPO, repo_url).group(2)
+                    if repo_name not in repo_dict:
+                        repo_dict[repo_name] = ""
+                        last = repo_name
+                    else:
+                        last = ""
+                else:
+                    last = ""
+            if raw_repo.startswith("    version: "):
+                repo_branch = raw_repo.replace("    version: ", "")
+                if last != "":
+                    repo_dict[last] = repo_branch
+                    last = ""
+                    repo_dict = recursive_get_repos_from_rosinstall(
+                        repo_owner, repo_name, repo_branch, repo_dict)
+    return repo_dict
+
+
+def format_message(data_dict):
```

Comment:
I would change the name of this function as it is only compatible with a specific type of message